### ISSUE TITLE
When using imageData, checkTesseractPresence is not called

### ISSUE THREAD

**Issue by samwilson** [thiagoalessio/tesseract-ocr-for-php#210]
### Expected behavior

The `run()` method should throw the following error for an unknown executable even when using `imageData()` to set the image.

> The command "foobar" was not found.

### Actual behavior

The actual error that is shown is:

> Error! Reading image data from stdin is not available this tesseract version
> Required version is 3.03-rc1, actual version is 1:
> 
> Generated command:
> "foobar" "" "/tmp/ocrKnFVot"

This is because `FriendlyErrors::checkTesseractVersion()` is called in `imageData()`, which happens before `FriendlyErrors::checkTesseractPresence()` is called in `run()`.

### Steps to reproduce the behavior

```php
$ocr = new TesseractOCR();
$ocr->executable('foobar');
$ocr->imageData($data, $size);
$ocr->run();
```
A workaround is to do this:
```php
$ocr = new TesseractOCR();
$ocr->executable('foobar');
FriendlyErrors::checkTesseractPresence($ocr->command->executable);
$ocr->imageData($data, $size);
$ocr->run();
```

#### Environment

* Operating system: `Linux 5.8.0-48-generic #54~20.04.1-Ubuntu SMP Sat Mar 20 13:40:25 UTC 2021 x86_64`
* PHP version: 7.4.3
* Tesseract version: 4.1.1
* tesseract-ocr-for-php version: 2.11.0

**Comment by thiagoalessio** [comment]
@samwilson thank you so much for reporting it. What you found is also true for other methods that also check tesseractVersion before checking the executable's presence. I'll let you know as soon as I push the fix. Thanks again.

**Comment by thiagoalessio** [comment]
@samwilson fix should be available in version [2.11.2](https://packagist.org/packages/thiagoalessio/tesseract_ocr#2.11.2). Feel free to re-open this issue in case the fix isn't complete, and thanks again for taking the time to report this bug!

**Comment by samwilson** [comment]
Terrific, thank you for the speedy response! :-)
